packages/insomnia-app/app/ui/redux/__tests__/selectors.test.ts

```diff
@@ -44,4 +46,104 @@ describe('selectors', () => {
       expect(project).toStrictEqual(expect.objectContaining<Partial<Project>>({ _id: DEFAULT_PROJECT_ID }));
     });
   });
+
+  describe('selectActiveApiSpec', () => {
+    it('will return undefined when there is not an active workspace', async () => {
+      const state = await reduxStateForTest({
+        activeWorkspaceId: null,
+      });
+
+      expect(selectActiveApiSpec(state)).toBe(undefined);
+    });
+
+    it('will return throw when there is not an active apiSpec', async () => {
+      const workspace = await models.workspace.create({
+        name: 'workspace.name',
+        scope: WorkspaceScopeKeys.design,
+      });
+
+      const state = await reduxStateForTest({
+        activeActivity: ACTIVITY_DEBUG,
+        activeWorkspaceId: workspace._id,
+      });
+      state.entities.apiSpecs = {};
+
+      const execute = () => selectActiveApiSpec(state);
+      expect(execute).toThrowError(`an api spec not found for the workspace ${workspace._id} (workspace.name)`);
+    });
+
+    it('will return the apiSpec for a given workspace', async () => {
+      const workspace = await models.workspace.create({
+        name: 'workspace.name',
+        scope: WorkspaceScopeKeys.design,
+      });
+      await models.apiSpec.updateOrCreateForParentId(
+        workspace._id,
+        { fileName: 'apiSpec.fileName' },
+      );
+
+      const state = await reduxStateForTest({
+        activeActivity: ACTIVITY_DEBUG,
+        activeWorkspaceId: workspace._id,
+      });
+
+      expect(selectActiveApiSpec(state)).toMatchObject({
+        contentType: 'yaml',
+        contents: '',
+        fileName: 'apiSpec.fileName',
+        parentId: workspace._id,
+        type: 'ApiSpec',
+      });
+    });
```

Comment:
Curious: why not store the result of `updateOrCreateForParentId` and check for the entire object here?

Conceptually it tells you that the object created above is the very object that should be in the assertion list. It connects the setup + assertion really explicitly.

```suggestion
    it('will return the apiSpec for a given workspace', async () => {
      const workspace = await models.workspace.create({
        name: 'workspace.name',
        scope: WorkspaceScopeKeys.design,
      });
      const spec = await models.apiSpec.updateOrCreateForParentId(
        workspace._id,
        { fileName: 'apiSpec.fileName' },
      );

      const state = await reduxStateForTest({
        activeActivity: ACTIVITY_DEBUG,
        activeWorkspaceId: workspace._id,
      });

      expect(selectActiveApiSpec(state)).toEqual(spec);
    });
```